### PR TITLE
Update contributor guidelines

### PR DESCRIPTION
**Pull request by PGijsbers**
Expanded and updated the contributor guide.
Some of this should additionally also be added to issue templates. 
It's probably best to do that at a GitHub organisational level (they will then apply to all projects, though they may override them).

### CODE REVIEW

File: docs/contributing.md
Comment (jsmatias) [inline docs/contributing.md:55]: What does the suggested...*

File: docs/contributing.md
Comment (jsmatias) [inline docs/contributing.md:57]: ... other existing ...

File: docs/contributing.md
Comment (jsmatias) [inline docs/contributing.md:59]: ... keep the scope ...

**Review by jsmatias — CHANGES_REQUESTED** [review]
Very good! Just added some minor corrections. After that it can be merged.

**Comment by PGijsbers** [comment]
Thanks, updated. Because the previous review had `requested changes` I cannot merge this unless I explicitly bypass branch protections. So an explicit approval would be nice.

**Pull request merged**
